projects/plugins/jetpack/modules/contact-form/grunion-contact-form.php

```diff
@@ -3191,7 +3456,7 @@ function process_submission() {
 
 		if ( ! $redirect ) { // wp_get_referer() returns false if the referer is the same as the current page.
 			$custom_redirect = false;
-			$redirect        = $_SERVER['REQUEST_URI'];
+			$redirect        = isset( $_SERVER['REQUEST_URI'] ) ? sanitize_text_field( wp_unslash( $_SERVER['REQUEST_URI'] ) ) : '';
```

Comment:
`sanitize_text_field` could easily mangle the URI by deleting percent-encoded characters.

Reply:
For that matter, it would probably be worth double-checking anything else that seems like it might be a URI. Like any "redirect_to" fields.

Reply:
Wanna just remove the `sanitize_text_field` and slap an ignore on it, or is there another method of sanitization we should use?

Reply:
`esc_url_raw` should work here. It has an alias, `sanitize_url`, that sounds more correct but the sniff doesn't recognize it (sigh).

Reply:
Tried to grab this and all the below in 305735a34b61aa4686458bff237ce38798da1c85